fud/fud/stages/verilator/tests/numeric_types.py

```diff
@@ -19,7 +19,7 @@ def test_fixed_point_round_trip(bits, is_signed):
     unsigned_integer = int(bit_string, 2)
     hex_string = np.base_repr(unsigned_integer, 16)
 
-    def fp_round_trip(bit_string: str) -> int:
+    def fp_round_trip(bit_string: str) -> FixedPoint:
```

Comment:
The return type for this is void right? Similar for `bitnum_round_trip`.

Reply:
Both the functions return a value so they can't have a void return type.

Reply:
```
    assert all(
        (
            bit_string == round_trip.bit_string(with_prefix=False),
            hex_string == round_trip.hex_string(with_prefix=False),
            unsigned_integer == round_trip.unsigned_integer(),
        )
    ), f"""width: {width}, is_signed: {is_signed}, bits: {bit_string}
        base 2: {bit_string} versus {round_trip.bit_string(with_prefix=False)}
        base 16: {hex_string} versus {round_trip.hex_string(with_prefix=False)}
        base 10: {unsigned_integer} versus {round_trip.unsigned_integer()}"""
```

They just throw an assertion upon failure. Otherwise, do nothing.

Reply:
You're reading the code for the surrounding function. The line above the you pasted uses the value returned frmo `fp_round_trip`.

Reply:
Ugh yup you're right. Belay my last.